### DISCUSSION TITLE
Scan sampling

### DISCUSSION THREAD

**Discussion by Sarah230323** [olivierfriard/BORIS#600] [Q&A]
Hi everyone,
I was wondering if it is possible to use scan sampling within an observation from media file(s) or just in live observation?

Thanks, Sarah

**Comment by olivierfriard** [comment]
Hi,

at this moment the option for scan sampling is only available for live observations.
I am working to add it to the media observations too.

Olivier

**Comment by olivierfriard** [comment]
Hi,

the scan sampling for media observations function is available in v. 8.18 as a pre-release (https://github.com/olivierfriard/BORIS/releases/tag/v8.18).
Please test it and let me know if it works as expected and if you have some suggestions.
Thank you for your collaboration

**Reply by Sarah230323** [reply]
Hi, 
Thank you! I'm lookong forward to try it and will let you know how my experience was. 
Sarah

**Reply by tksidhu** [reply]
Hi Olivier,

I'm trying to use this feature but it keeps crashing and the time offset isn't quite right. Here is the error report:
BORIS version: 8.18
OS: Windows 10 10.0.22621
CPU: AMD64 Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
Python 3.11.2 (64-bit)
Qt 5.15.2 - PyQt 5.15.9
2023-05-10 18:03

Traceback (most recent call last):
  File "C:\Users\Psychology\BORIS\Lib\site-packages\boris\core.py", line 3709, in mpv_timer_out
    while self.media_scan_sampling_mem[-1] > cumulative_time_pos:
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^
IndexError: list index out of range

System info:

Host Name:                 DESKTOP-IUFJ8CB
OS Name:                   Microsoft Windows 11 Enterprise
OS Version:                10.0.22621 N/A Build 22621
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          Psychology
Registered Organization:   HP
Product ID:                00329-10180-00000-AA075
Original Install Date:     13/11/2022, 21:41:36
System Boot Time:          09/05/2023, 23:05:15
System Manufacturer:       HP
System Model:              HP Spectre x360 Convertible 15-eb0xxx
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 165 Stepping 2 GenuineIntel ~2592 Mhz
BIOS Version:              AMI F.14, 06/04/2022
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-gb;English (United Kingdom)
Input Locale:              en-gb;English (United Kingdom)
Time Zone:                 (UTC+00:00) Dublin, Edinburgh, Lisbon, London
Total Physical Memory:     16,088 MB
Available Physical Memory: 5,102 MB
Virtual Memory: Max Size:  31,448 MB
Virtual Memory: Available: 12,701 MB
Virtual Memory: In Use:    18,747 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\DESKTOP-IUFJ8CB
Hotfix(s):                 5 Hotfix(s) Installed.
                           [01]: KB5022497
                           [02]: KB5012170
                           [03]: KB5026372
                           [04]: KB5025749
                           [05]: KB5025351
Network Card(s):           4 NIC(s) Installed.
                           [01]: ExpressVPN TAP Adapter
                                 Connection Name: Ethernet 2
                                 Status:          Media disconnected
                           [02]: ExpressVPN Wintun Driver
                                 Connection Name: Local Area Connection
                                 Status:          Media disconnected
                           [03]: Intel(R) Wi-Fi 6 AX201 160MHz
                                 Connection Name: WiFi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     148.197.10.226
                                 IP address(es)
                                 [01]: 10.128.32.20
                                 [02]: fe80::7b94:767f:c21f:54a1
                           [04]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
Hyper-V Requirements:      VM Monitor Mode Extensions: Yes
                           Virtualization Enabled In Firmware: Yes
                           Second Level Address Translation: Yes
                           Data Execution Prevention Available: Yes

**Comment by olivierfriard** [comment]
Hi,

I am not able to reproduce the problem.
Can you give me more information about your observation ?
What are the values of the parameters you used?

**Comment by tksidhu** [comment]
Hi, here is a screenshot of the observation parameters. I found it works
fine for the first interval and then after I press play again it stops and
beeps less than every second and crashes.

[image: image.png]

**Reply by tksidhu** [reply]
![image](https://github.com/olivierfriard/BORIS/assets/61194642/7806809f-abd1-47fc-af2f-9f0f71d6be22)

**Comment by olivierfriard** [comment]
You have set an observation interval from 55.683 to 61.094. 
Are you sure that the observation has not reached the end of the interval (61.094) and BORIS is beeping for that?

**Reply by tksidhu** [reply]
Yes sorry there was a typo. However, I have another question about the image display duration. I assumed the function of this would be either 
a) how long the media continues to play during the scan sampling interval (i.e., say I wanted to analyse a 1 second clip of my media every 15 seconds)
or b) how long the player remains paused on the frame until it starts playing again
Neither of these seem to be the case... I can't find any information in the manual - could you explain further please?

**Comment by olivierfriard** [comment]
The manual is not updated to the last version (no time to do that now).
The image display duration option is not correlated to scan sampling (even if the two options are on the same row...) 
The image display duration applies in case of a mixed observation (video and images). In this case BORIS will display the images for the time you selected.

In your case, BORIS will pause the video every 15 s and you will have to restart the video playing manually.
To observe for 1 second you can use the frame-by-frame mode (left/right arrow keys)

**Reply by tksidhu** [reply]
Got it. Thank you so much for taking the time to reply!
